test/pkg/debug/node.go

```diff
@@ -79,6 +80,6 @@ func (c *NodeController) Register(ctx context.Context, m manager.Manager) error
 				return o.GetLabels()[karpv1.NodePoolLabelKey] != ""
 			}),
 		)).
-		WithOptions(controller.Options{MaxConcurrentReconciles: 10}).
+		WithOptions(controller.Options{MaxConcurrentReconciles: 10, SkipNameValidation: lo.ToPtr(true)}).
```

Comment:
Do we actually want `SkipNameValidation` here (and below?). Shouldn't we be having unique controller names?

Reply:
These controllers are only injected in testing, so probably not a big deal, but I don't see why they can't be named uniquely, will rename instead.

Reply:
The tests create a new manager with these debug controllers in BeforeEach, but controller-runtime tracks all controller names ever observed, preventing reuse even between managers (https://github.com/kubernetes-sigs/controller-runtime/issues/2937), producing "Controller names must be unique to avoid multiple controllers reporting the same metric" on attempted reuse. Disabling name validation is the recommended approach for tests. There is a way to set this at the manager level, but leaving it in the debug controllers seems better, so leaving as is.